### DISCUSSION TITLE
Is functional update available in Jotai?

### DISCUSSION THREAD

**Discussion by a-oka-z** [pmndrs/jotai#1371] [Q&A]
Hello. At first I'd like to say my appreciation for developing such a great library. Thank you 😃

I have a question; is functional update like useState in React available in Jotai? 

```javascript
// Copied from https://reactjs.org/docs/hooks-reference.html#usestate
function Counter({initialCount}) {
  const [count, setCount] = useState(initialCount);
  return (
    <>
      Count: {count}
      <button onClick={() => setCount(initialCount)}>Reset</button>
      <button onClick={() => setCount(prevCount => prevCount - 1)}>-</button>
      <button onClick={() => setCount(prevCount => prevCount + 1)}>+</button>
    </>
  );
}
```
Sometimes I would like to update an atom to the value which depends on its previous state; but in examples in Jotai's documentation they are simply referring their current values.

In such situation, do I have to write Jotai's read-only/write-only atoms? Or is there something provided for such situation? I know I have something missing but I still don't know what it is.

Thank you.

**Comment by dai-shi** [comment]
Yes, function update just works for primitive atoms like atom(0).
